src/Sylius/Bundle/CoreBundle/CatalogPromotion/CommandHandler/ArchiveCatalogPromotionHandler.php

```
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler;
+
+use DateTimeImmutable;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\ArchiveCatalogPromotion;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Processor\CatalogPromotionStateProcessorInterface;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Promotion\Exception\InvalidCatalogPromotionStateException;
+use Sylius\Component\Promotion\Model\CatalogPromotionStates;
+use Sylius\Component\Promotion\Repository\CatalogPromotionRepositoryInterface;
+
+final class ArchiveCatalogPromotionHandler
+{
+    public function __construct(
+        /** @var CatalogPromotionRepositoryInterface<CatalogPromotionInterface> */
+        private CatalogPromotionRepositoryInterface $catalogPromotionRepository,
+        private CatalogPromotionStateProcessorInterface $catalogPromotionStateProcessor,
+    ) {
+    }
+
+    public function __invoke(ArchiveCatalogPromotion $command): void
+    {
+        /** @var CatalogPromotionInterface|null $catalogPromotion */
+        $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $command->code]);
+
+        if (null === $catalogPromotion) {
```

I'm unaware if the new feature catalog promotion code-base, but usually this should be asserted as command should fail explicitly.

---

That's copied from other handlers. I would let this stay as it is, to be consistent.